### PR TITLE
Added datasources but no link

### PR DESCRIPTION
**Pull request by jvwong**
re #1082 adding datasources for interactions but no link.

### CODE REVIEW

**Comment by jvwong** [comment]
example for tp53 mdm2

image

File: src/client/features/interactions/interactions-edge-tooltip.js [hunk 1]
```diff
@@ -10,8 +10,17 @@ class InteractionsEdgeTooltip extends React.Component {
     super(props);
 
     this.state = {
+      datasources: [],
       publications: []
     };
+
+    // Update datasources content to get homepage?
+    ServerAPI.datasources()
```
Comment (d2fong) [inline src/client/features/interactions/interactions-edge-tooltip.js]: Why can't you send this request once at the time the network is generated and compute the datasources for all elements there?

File: src/client/features/interactions/interactions-edge-tooltip.js [hunk 2]
```diff
@@ -24,11 +33,22 @@ class InteractionsEdgeTooltip extends React.Component {
...
   render(){
+    const state = this.state;
     let { edge } = this.props;
     let { publications } = this.state;
     let title = edge.data('id');
+    let providers = edge.data('providers');
     let pcIds = edge.data('pcIds');
 
+    let providersList = providers.map( provider => {
+      const dsInfo = _.isEmpty(state.dataSources) ? {}: _.find(state.dataSources, ds => {
```
Comment (d2fong) [inline src/client/features/interactions/interactions-edge-tooltip.js]: I don't like introducing another instance of the 'find datasource by id' problem into another area of the codebase.  We need to come up with a solution to keep this logic in one place and make it reusable.

File: src/server/graph-generation/interaction/index.js [hunk 3]
```diff
@@ -48,7 +50,9 @@ let interactionTxt2CyJson = (srcId, tgtId, type, pubmedIdsString, mediatorIdsStr
       type,
       source: srcId,
       target: tgtId,
+      providers,
```
Comment (d2fong) [inline src/server/graph-generation/interaction/index.js]: You need to update the tests.

**Comment by jvwong** [comment]
ok yeah better you than me. I was having a hard time figuring out how to do all that. THanks!

**Pull request merged**